### PR TITLE
Format all code with clang-format

### PR DESCRIPTION
**Pull request by Kozoriz**
* Add python scripts to run clang-format
* Format all project code except 3rd parties with clang-format 3.6
* Add pre-commit hook to check coding style

### CODE REVIEW

**Comment by Kozoriz** [comment]
Please review : @anosach-luxoft , @dev-gh , @AByzhynar , @VProdanov

**Comment by anosach-luxoft** [comment]
Reviewed. Please do not use `+/-` symbols in commit messages. It's meanless and looks not good.

**Comment by AByzhynar** [comment]
Reviewed

**Comment by dev-gh** [comment]
Visually reviewed

**Comment by VProdanov** [comment]
Reviewed

**Comment by malirod** [comment]
Reviewed

File: COMMITTERS.md [hunk 1]
```diff
@@ -18,6 +18,12 @@ set in the form of a pull request.
 
 **develop branch** - [The branch](https://github.com/LuxoftSDL/sdl_core/tree/develop) where bug fixes against the latest release or release candidate are merged.
 
+## Pre-commit hook instalation
```
Comment (anosach-luxoft) [inline COMMITTERS.md]: @Kozoriz typo `instalation`

Comment (Kozoriz) [inline COMMITTERS.md]: @anosach-luxoft , done e316d04

**Comment by LuxoftAKutsan** [comment]
@Kozoriz  Why I wasn' t invited in this pull request?

**Pull request merged**